admin/services/class-file-size.php

```diff
@@ -0,0 +1,88 @@
+<?php
+/**
+ * WPSEO plugin file.
+ *
+ * @package WPSEO\Admin\Services
+ */
+
+/**
+ * Represents the file size service.
+ */
+class WPSEO_File_Size_Service {
+
+	/**
+	 * Retrieves an indexable.
+	 *
+	 * @param WP_REST_Request $request The request object.
+	 *
+	 * @return WP_REST_Response The response.
+	 */
+	public function get( WP_REST_Request $request ) {
+		$file_url = rawurldecode( $request->get_param( 'url' ) );
+
+		if ( $this->is_externally_hosted( $file_url ) ) {
+			return new WP_REST_Response(
+				array(
```

Comment:
This feels like we're reinventing the wheel. I'd suggest using an exception for this instead.